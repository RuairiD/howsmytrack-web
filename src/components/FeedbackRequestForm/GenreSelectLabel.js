import React from "react";

import { Button, Icon, Tooltip } from "antd";
import { A, Span, Strong } from "lemon-reset";

const GenreTooltipContent = (
    <Span>
        Requests of the same genre are assigned to the same groups in order to get you the most relevant and constructive feedback. Don't worry if your genre isn't listed though; just submit under <Strong>No Genre</Strong> and you'll still be grouped! <A target="_blank" rel="noopener noreferrer" href="/faq#why-is-my-tracks-genre-not-listed">Why is my genre not listed?</A>
    </Span>
);

const GenreSelectLabel = () => (
    <React.Fragment>
        Genre :
        <Tooltip
            title={GenreTooltipContent}
        >
            <Button type="link"><Icon type="question-circle" />What is this for?</Button>
        </Tooltip>
    </React.Fragment>
);

export default GenreSelectLabel;
