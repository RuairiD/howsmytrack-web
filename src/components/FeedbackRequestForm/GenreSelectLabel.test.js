import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import GenreSelectLabel from "./GenreSelectLabel";

describe("GenreSelectLabel", () => {
    it("renders a label for a GenreSelect, including informational tooltip", () => {
        const wrapper = shallow(
            <GenreSelectLabel />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
