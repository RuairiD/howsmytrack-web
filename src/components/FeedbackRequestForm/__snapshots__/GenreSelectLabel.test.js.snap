// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreSelectLabel renders a label for a GenreSelect, including informational tooltip 1`] = `
<Fragment>
  Genre :
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="top"
    title={
      <Span
        className=""
      >
        Requests of the same genre are assigned to the same groups in order to get you the most relevant and constructive feedback. Don't worry if your genre isn't listed though; just submit under 
        <Strong
          className=""
        >
          No Genre
        </Strong>
         and you'll still be grouped! 
        <A
          className=""
          href="/faq#why-is-my-tracks-genre-not-listed"
          rel="noopener noreferrer"
          target="_blank"
        >
          Why is my genre not listed?
        </A>
      </Span>
    }
    transitionName="zoom-big-fast"
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      type="link"
    >
      <Icon
        type="question-circle"
      />
      What is this for?
    </Button>
  </Tooltip>
</Fragment>
`;
